doc-deploy-dev-index/action.yml

```diff
@@ -0,0 +1,139 @@
+name: >
+  Doc deploy dev index action
+
+description: |
+  This action automates the process of creating indexes and scraping the
+  desired HTML documentation artifact that contains the development version
+  of a library, then deploying it to a Meilisearch instance.
+
+inputs:
+
+  # Required inputs
+
+  cname:
+    description: >
+      The Canonical Name (CNAME) that points to the documentation website.
+    required: true
+    type: string
+
+  library-name:
+    description: >
+      The name of the library.
+    required: true
+    type: string
+
+  api-key:
+    description: >
+      The API key used to access the Meilsearch instance host.
+    required: true
+    type: string
+
+  host-url:
+    description: >
+      The URL where the Meilsearch instance is hosted.
+    required: true
+    type: string
+
+  # Optional inputs
+
+  doc-artifact-name:
+    description: >
+      The name of the HTML documentation artifact. This artifact is expected to
+      contain all the HTML and static files.The dafault value is ``documentation-html``.
+    required: false
+    default: 'documentation-html'
+    type: string
+
+  template:
+    description: >
+      The "template" parameter specifies the layout used for the HTML documentation.
+      By default, it is set to ``sphinx_pydata`` which assumes that the document is
+      constructed using the pydata-sphinx-theme or its associated theme,
+      such as the ansys-sphinx-theme.
+    required: false
+    default: "sphinx_pydata"
+    type: string
+
+
+
+runs:
+  using: "composite"
+  steps:
+
+    # ------------------------------------------------------------------------
+
+    - uses: ansys/actions/_logging@main
+      with:
+        level: "INFO"
+        message: >
+          Checkout the repository branch for deploying the documentation. If
+          this step fails, then it means that the provided token is not valid.
+
+    - name: "Get the name of the repository"
+      shell: bash
+      run: |
+        if [[ "${{ inputs.repository }}" == "current" ]]; then
+          echo "REPOSITORY=${{ github.repository }}" >> $GITHUB_ENV
+        else
+          echo "REPOSITORY=${{ inputs.repository }}" >> $GITHUB_ENV
+        fi
+
+    - name: "Checkout ${{ env.REPOSITORY }} repository"
+      uses: actions/checkout@v3
+      with:
+        repository: ${{ env.REPOSITORY }}
+        ref: ${{ inputs.branch }}
+        token: ${{ inputs.token }}
```

Comment:
Why checking the repository? No need to clone the repo. We should only install ``pymeilisearch`` and upload the indexed artifacts to the instance.